src/cli/main.ts

```diff
@@ -10,81 +10,86 @@ import { rootLogger } from '../lib/nexus-logger'
 import { detectExecLayout, globalLocalHandoff } from '../lib/process'
 import * as Commands from './commands'
 
-const log = rootLogger
+main()
 
-// use envar to boost perf, skip costly detection work
-if (!process.env.GLOBAL_LOCAL_HANDOFF) {
-  const depName = 'nexus'
+function main() {
+  const log = rootLogger
 
-  log.trace('execLayout start')
-  const execLayout = detectExecLayout({ depName })
-  log.trace('execLayout done', { execLayout })
+  // use envar to boost perf, skip costly detection work
+  if (!process.env.GLOBAL_LOCAL_HANDOFF) {
+    const depName = 'nexus'
 
-  if (execLayout.toolProject && !execLayout.runningLocalTool) {
-    if (execLayout.toolCurrentlyPresentInNodeModules) {
-      globalLocalHandoff({
-        localPackageDir: path.join(execLayout.project!.nodeModulesDir, depName),
-        globalPackageFilename: __filename,
-      })
-    } else {
-      //todo detect package manager
-      // const packageManager = await createPackageManager(undefined, { projectRoot })
-      log.fatal(
-        stripIndent`
+    log.trace('execLayout start')
+    const execLayout = detectExecLayout({ depName })
+    log.trace('execLayout done', { execLayout })
+
+    if (execLayout.toolProject && !execLayout.runningLocalTool) {
+      if (execLayout.toolCurrentlyPresentInNodeModules) {
+        globalLocalHandoff({
+          localPackageDir: path.join(execLayout.project!.nodeModulesDir, depName),
+          globalPackageFilename: __filename,
+        })
+        return
```

Comment:
This is the the only real change.

I wrapped the scrip into a `main` func so the diff looks bigger than it is.

Reply:
I decided to refactor and get us close a point where we can extract to `@labs/glocal`.

I realized that while the system etc. tests may be valid we should still have better isolated testing. And i realized that there is actually still missing scope with this utility, which now has a proper name (`glocal`).

Reply:
Open to discussion but I'd like not to split up everything as packages as much as possible unless we want to re-use that code somewhere else. If we want to split everything as packages, then I think we should have a mono-repo (which, I know, dripip doesn't support) otherwise it makes even small updates a pain in the ass, cause you have to publish 12 packages when doing refactors across your codebase.

Keeping things separate in a `lib` folder just like it is right now works pretty well I think, especially for very small libs like this one

Reply:
These are soft points but other benefits can include easier code navigation, easier testing, easier enforcement of encapsulation. It is also a risk to maintain a lot of project-proprietary code. At least as an external package there is potential for it to catch its own wind and life.

I agree with the downsides you point out. I think they can be mitigated by "graduating" packages that are stable enough. I think it's fair to say `glocal` isn't there yet, since its brand new really.

I think points on both sides can be seen as solvable problems by better tooling. I think its accurate to say the tooling for wrangling many packages, today, and probably ever, is not so good as to make it as easy as just-another-module within the monolithic source.